scripts/tools/DeleteDeliveryScript.php

```diff
@@ -18,66 +32,75 @@
  * sudo -u www-data php index.php 'oat\taoDeliveryRdf\scripts\tools\DeleteDeliveryScript'
  *
  * Class TerminateSession
+ *
  * @package oat\taoDeliveryRdf\scripts\tools
  */
 class DeleteDeliveryScript extends ScriptAction
 {
-    /**
-     * @var common_report_Report
-     */
-    private $report;
-
-    protected function showTime()
+    protected function showTime(): bool
     {
         return true;
     }
 
-    protected function provideUsage()
+    protected function provideUsage(): array
     {
         return [
-            'prefix' => 'h',
-            'longPrefix' => 'help',
-            'description' => 'Prints a help statement'
+            'prefix'      => 'h',
+            'longPrefix'  => 'help',
+            'description' => 'Prints a help statement',
         ];
     }
 
-    protected function provideOptions()
+    protected function provideOptions(): array
     {
         return [
-            'delivery' => [
-                'prefix' => 'd',
-                'longPrefix' => 'delivery',
-                'required' => true,
-                'description' => 'A delivery ID.'
-            ]
+            'delivery'  => [
+                'prefix'      => 'd',
+                'longPrefix'  => 'delivery',
+                'required'    => true,
+                'description' => 'A delivery ID',
+            ],
+            'recursive' => [
+                'prefix'      => 'r',
+                'longPrefix'  => 'recursive',
+                'flag'        => true,
+                'description' => 'Remove all the linked resources such as Tests or Items',
+            ],
         ];
     }
 
-    protected function provideDescription()
+    protected function provideDescription(): string
     {
         return 'TAO Delivery - Delete Delivery';
     }
 
-    /**
-     * @return Report
-     * @throws \common_exception_Error
-     */
-    protected function run()
+    protected function run(): Report
     {
-        $this->report = common_report_Report::createInfo('Deleting Delivery ...');
+        $report = Report::createInfo('Deleting Delivery ...');
```

Comment:
could we remove the static call ?

Reply:
Could you please be more specific?
I don't get your point.

Reply:
yes :) , i would inject Report and then just $report->add()
just to be able to not use the static call and to be able to 
substitute i would actually strong type it with ReportInterface
but this is not a blocker in your PR 
it is just a preference to clean up the static calls in new code